xcat-inventory/xcclient/allien/regexutils.py

```diff
@@ -0,0 +1,95 @@
+###############################################################################
+# IBM(c) 2019 EPL license http://www.eclipse.org/legal/epl-v10.html
+###############################################################################
+
+# -*- coding: utf-8 -*-
+
+import re
+
+def extract_bracketed(input_str):
+    """extract bracketed expression into dict
+       input_str example: 10.($1+1).0.($2+3)
+       output example:
+        { 'prev': '10.',
+          'curr': '($1+1)',
+          'next': '.0.($2+3)'
+        }
```

Comment:
The method might be wrong when there are embeded bracked, for example,
```
extract_bracketed("10.($1+$2-1+($2/3)).0.($3)")
{'prev': '10.', 'curr': '($1+$2-1+($2/3)', 'next': ').0.($3)'}
```

Not sure if this is right behavior.